### PR TITLE
Add log level for internal log of sdk

### LINKED ISSUE #1138: Could internal log level be set by runtime?
**Is your feature request related to a problem?**

We usually need debug log when testing and integrating opentelemetry-cpp, and just need error log after that.But now the internal log  level can only be set when compiling, which is usual WARNING.

**Describe the solution you'd like**

Maybe add `GlobalLogHandler::SetLogLevel(...)` or add `log_level` to `GlobalLogHandler::SetLogHandler(...)` ?

### PR DESCRIPTION
**Pull request by owent**
Signed-off-by: owentou <owentou@tencent.com>

Fixes #1138 

## Changes

+ Add log level for internal log of sdk
+ Fix document for internal log of sdk
+ Fix crash problem when any provider is destroyed after internal log handle(Both are signaltons).
  > It may call `OTEL_INTERNAL_LOG_*(...)` and will crash.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: sdk/include/opentelemetry/sdk/common/global_log_handler.h [hunk 1]
```diff
@@ -136,74 +147,71 @@ class GlobalLogHandler
...
+      break;                                                                              \
+    }                                                                                     \
+    std::stringstream tmp_stream;                                                         \
+    tmp_stream << message;                                                                \
+    log_handler->Handle(level, __FILE__, __LINE__, tmp_stream.str().c_str(), attributes); \
+  } while (false);
 
 #define OTEL_INTERNAL_LOG_GET_3RD_ARG(arg1, arg2, arg3, ...) arg3
 
-#if OTEL_INTERNAL_LOG_LEVEL >= OTEL_INTERNAL_LOG_LEVEL_ERROR
```
Comment (lalitb) [inline sdk/include/opentelemetry/sdk/common/global_log_handler.h:151]: Do you think we should keep OTEL_INTERNAL_LOG_LEVEL along with the run-time level? If this macro is defined, it should override the run-time levels ( it setting run-time level will not have any effect). One of the advantages of having a compile-time macro OTEL_INTERNAL_LOG_LEVEL is that macro-call `OTEL_INTERNAL_LOG_DISPATCH` could be totally removed from the hot-path of the event pipeline based on the level defined. We can let it not be defined by default in our build, but users have the option to enable it if they want.

Comment (owent) [reply sdk/include/opentelemetry/sdk/common/global_log_handler.h:151]: Yes, of course. Can we define the use `OTEL_INTERNAL_LOG_LEVEL_DEBUG` as the default value of `OTEL_INTERNAL_LOG_LEVEL` ? Many users use the prebuilt packages provided by package manager and need debug log to test integration.

Comment (lalitb) [reply sdk/include/opentelemetry/sdk/common/global_log_handler.h]: Yes, that would be good too. Thanks.

**Review by lalitb — APPROVED** [review]
Thanks

File: sdk/include/opentelemetry/sdk/common/global_log_handler.h [hunk 2]
```diff
@@ -16,7 +17,7 @@
 #define OTEL_INTERNAL_LOG_LEVEL_DEBUG 3  // to be disabled in release
 
 #ifndef OTEL_INTERNAL_LOG_LEVEL
-#  define OTEL_INTERNAL_LOG_LEVEL OTEL_INTERNAL_LOG_LEVEL_WARN  // ERROR and WARN
+#  define OTEL_INTERNAL_LOG_LEVEL OTEL_INTERNAL_LOG_LEVEL_DEBUG  // ERROR and WARN
```
Comment (lalitb) [inline sdk/include/opentelemetry/sdk/common/global_log_handler.h:20]: nit  - change the comment?

Comment (owent) [reply sdk/include/opentelemetry/sdk/common/global_log_handler.h]: > nit - change the comment?

done

**Comment by lalitb** [comment]
@owent - this looks good to merge. Would you rebase it to the main branch? Thanks.

**Comment by owent** [comment]
> @owent - this looks good to merge. Would you rebase it to the main branch? Thanks.

Done

**Pull request merged**